pygen/pygen_src/riscv_instr_gen_config.py

```diff
@@ -411,11 +411,13 @@ def parse_args():
     parse.add_argument('--asm_test_suffix', help = 'asm_test_suffix', default = "")
     parse.add_argument('--march_isa', help = 'march_isa', default = [],
                        choices = [i.name for i in riscv_instr_group_t], nargs = '*')
-    parse.add_argument('--directed_instr_0', help = 'directed_instr_0',
-                       default = "riscv_int_numeric_corner_stream,4")
-    parse.add_argument('--stream_name_opts', help = 'stream_name_0',
-                       default = "riscv_load_store_rand_instr_stream")
-    parse.add_argument('--stream_freq_opts', help = 'stream_freq_0', default = 4)
+    for i in range(7):
```

Comment:
Ideally this should be `for i in range(self.max_directed_instr_stream_seq)` to allow for custom amounts of directed sequences (what if the user inputs more than 7 directed streams?).
To do this though, `parse_args` would need to be moved inside of `riscv_instr_gen_config` object - is this possible?
Possibly you could call this `parse_args` function and do all the `args_dict` setup at the end of `riscv_instr_gen_config.__init__()` .

Reply:
Yes, It is possible. Will fix this